### PR TITLE
fix: update user task when formId is not set during importing

### LINKED ISSUE #24771: User Tasks with linked forms started with the WebModeler's `Play` not being persisted correctly
### Describe the bug

When deploying a process with a linked form using the WebModeler's `Play` functionality, occasionally the created User Tasks do not have correct reference on the linked form. 

### To Reproduce

1. Create a form in the WebModeler but do not deploy it
2. Create a simple BPMN process with a User Task and link it to the previous form
3. Use WebModeler's `Play` functionality to deploy and run all components

### Actual behavior
Sometimes, the fields `formId` and `formVersion` are not being included in the User Task flushed record. Persisting only the `formKey` results in Tasklist OOTB FE to not be able to fetch the form, thus displaying `Task has no variables` message.

### Expected behavior
The `formId` and `formVersion` are always populated in the User Task record. Tasklist OOTB can retrieve the relevant form.

### Environment

- OS: [SaaS]
- Browser: [e.g. chrome, safari]
- Tasklist Version: Verified on 8.6 and 8.5, reported also for 8.4 and 8.3

### Additional context
Reported on slack: https://camunda.slack.com/archives/C043W5V88M7/p1731394875251639

---------------------------------------------------------------------------------------------

### Definition of Ready - Checklist

- [ ] The bug has been reproduced by the assignee in the environment compatible with the provided one; otherwise, the issue is closed with a comment
- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] Necessary screenshots, screen recordings, or files are attached to the bug report

For UI changes required to solve the bug:

- [ ] Design input has been collected by the assignee

### After solving the issue - Answer and document in the comments the questions below

- [ ] What was the root cause for this bug?
- [ ] Is there anything we should do for the error to not happen anymore?

### PR DESCRIPTION
**Pull request by panagiotisgts**
## Description

Fix the issue that occurs when a Form is not present/indexed yet when a User Task is created.

When the UserTask is queried check if it's `formId` field is not set and if other conditions apply and perform an update on the UserTask setting the `formId`.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #24771

### CODE REVIEW

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/TaskService.java [hunk 1]
```diff
@@ -126,7 +129,21 @@ public List<TaskDTO> getTasks(
   }
 
   public TaskDTO getTask(final String taskId) {
-    return TaskDTO.createFrom(taskStore.getTask(taskId), objectMapper);
+    final TaskEntity task = taskStore.getTask(taskId);
+    if (task.getExternalFormReference() == null
+        && (task.getIsFormEmbedded() == null || task.getIsFormEmbedded())
+        && task.getFormKey() != null
+        && task.getFormId() == null) {
+      final Optional<FormIdView> linkedForm =
```
Comment (houssain-barouni) [inline tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/TaskService.java:137]: can you a log , it can be useful

**Review by houssain-barouni — APPROVED** [review]
LGTM

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/TaskService.java [hunk 2]
```diff
@@ -126,7 +130,24 @@ public List<TaskDTO> getTasks(
...
-    return TaskDTO.createFrom(taskStore.getTask(taskId), objectMapper);
+    final TaskEntity task = taskStore.getTask(taskId);
+    if (taskFormLinkIsNotComplete(task)) {
+      LOGGER.debug(
+          "Task with id {} found having incorrect form linking to form with key {}",
+          taskId,
+          task.getFormKey());
+
+      final Optional<FormIdView> linkedForm =
+          formStore.getHighestVersionFormByKey(task.getFormKey());
```
[the remark this answers is not in the document]
Comment (panagiotisgts) [reply tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/TaskService.java:141]: Yeah I thought of that as well... I'll not merge this one yet and do a quick investigation tomorrow if it makes sense to update UserTasks on Form import without needing a lot of changes.

[the remark this answers is not in the document]
Comment (panagiotisgts) [reply tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/TaskService.java:141]: @houssain-barouni  Thinking again about this, the `formKey` is unique for subsequent `formVersions` on the same `formId` isn't it? So technically you can never have a version mismatch when searching by the `formKey` ? :thinking:

**Review by houssain-barouni — APPROVED** [review]

**Pull request merged**